src/lib/udmp-parser.h

```diff
@@ -987,6 +1052,32 @@ class UserDumpParser {
     return ForegroundThreadId_;
   }
 
+  std::string to_string() const {
+    std::stringstream ss;
+    ss << "UserDumpParser(";
+    ss << "ModuleNb=" << Modules_.size();
+    ss << ", ThreadNb=" << Threads_.size();
+    ss << ")";
+    return ss.str();
+  }
+
+  std::vector<uint8_t> ReadMemory(uintptr_t Address, size_t Size) const {
+    std::vector<uint8_t> Out;
+
+    auto const Block = GetMemBlock(Address);
+    if (!Block) {
+      throw ParsingError(std::string("Invalid address"));
+    }
+
+    const uint64_t OffsetFromStart = Address - Block->BaseAddress;
+    const size_t RemainingSize =
+        Block->DataSize - static_cast<size_t>(OffsetFromStart);
+    const size_t DumpSize = std::min(RemainingSize, Size);
+    Out.resize(DumpSize);
```

Comment:
Actually you can probably directly instantiate `Out` here using the constructor that allocates `n` items directly?

Reply:
Good catch, done.